service/discovery/src/main/java/org/jolokia/service/discovery/DiscoveryMulticastResponder.java

```diff
@@ -20,40 +17,39 @@ public class DiscoveryMulticastResponder extends AbstractJolokiaService<JolokiaS
 
     // Listener threads responsible for creating the response as soon as a discovery request
     // arrives.
-    private final List<MulticastSocketListenerThread> listenerThreads;
+    private MulticastSocketListenerThread listenerThread;
```

Comment:
So you finally concluded that a single thread is enough for the class and that's why you removed the usage of List here?  Just checking.

Reply:
Exactly. I realized (with my friend debugger) that in my case where I have `eth0` and `tun0` interfaces, each with 2x IPv6 addresses and 1x IPv4 address, I had **six** `MulticastSocketListenerThread` threads, each listening on _reusable_ multicast UDP socket - each on **the same** `*:24884` interface:port.